tests/lax_numpy_test.py

```diff
@@ -594,6 +593,21 @@ def testBitwiseOp(self, np_op, jnp_op, rng_factory, shapes, dtypes):
                             check_dtypes=jtu.PYTHON_SCALAR_SHAPE not in shapes)
     self._CompileAndCheck(jnp_op, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": f"_op={op.__name__}_dtype={dtype.__name__}"
+                      f"_x_shape={x_shape}_y_shape={y_shape}",
+     "op": op, "dtype": dtype,
+     "x_shape": x_shape, "y_shape": y_shape, "rng_factory": rng_factory}
+    for op in [jnp.left_shift, jnp.right_shift]
+    for dtype in int_dtypes + unsigned_dtypes
+    for x_shape, y_shape in [((), (2, 3)), ((2, 3), ()), ((2, 3), (2, 3))]
+    for rng_factory in [jtu.rand_int]))
+  def testShiftOp(self, op, dtype, x_shape, y_shape, rng_factory):
+    rng = rng_factory(self.rng())
+    args_maker = lambda: (rng(x_shape, dtype), np.full(y_shape, 2, dtype))
+    self._CompileAndCheck(op, args_maker)
+    self._CheckAgainstNumpy(getattr(np, op.__name__), op, args_maker)
```

Comment:
This is a pretty standard test: I think it would be sufficient to use an `op_record` above.

Reply:
@jakevdp How do I get `np.full(y_shape, 2, dtype)` with `op_record`?

Reply:
I don't think we should only test a single value for the second entry. We should generate a full array of random values by the standard mechanism in order to get fuller test coverage.

Reply:
As I understand, we could use something like `jtu.rand_int(8)`, but how do we apply it only to the shift arg?

Reply:
I think this would do it:
```python
op_record("right_shift", 2, int_dtypes + unsigned_dtypes, all_shapes, jtu.rand_default, [])
```

Reply:
@jakevdp 

> I think we shouldn't use just jtu.rand_int, since this will shift values mostly out of bounds, almost never covering useful cases.

Reply:
By default, the integers in `rand_default` are generated from a roughly Gaussian distribution with a standard deviation of 3, so I think it will cover relevant values.

Reply:
Ah, ok!

Reply:
@jakevdp We can't use a record with `rand.default` for the shift arg because both NumPy and XLA require shifts to be positive and smaller than the bit width. I have reverted to writing a separate test.